### PR TITLE
Port to Gtk4

### LINKED ISSUE #144: Checkbox outline is cut off
### What Happened?

Checkbox outline is cut off

image

### Steps to Reproduce

1. Read readme.md or use Sideload

### Expected Behavior

Outline should not be cut off

Setting checkbox margin_bottom to 1 fixes this issue

### OS Version

6.x (Odin)

### Software Version

Compiled from git

### PR DESCRIPTION
**Pull request by danrabbit**
Fixes #144

Todo:
- [x] Cancel when mainwindow is closed
- [x] Notify when out of focus

### CODE REVIEW

File: src/Views/AbstractView.vala [hunk 1]
```diff
@@ -18,60 +18,74 @@
...
-        primary_label.xalign = 0;
-        primary_label.get_style_context ().add_class (Granite.STYLE_CLASS_PRIMARY_LABEL);
+        primary_label.add_css_class (Granite.STYLE_CLASS_TITLE_LABEL);
 
-        secondary_label = new Gtk.Label (null);
+        secondary_label = new Gtk.Label (null) {
+            max_width_chars = 50,
+            width_chars = 50,
+            xalign = 0
+        };
```
Comment (jeremypw) [inline src/Views/AbstractView.vala:57]: Any reason why other properties not initialised in constructor?

**Comment by jeremypw** [comment]
Some visual glitches noticed - not sure whether this is an upstream or gala issue though as I noticed similar problems with the screen not properly updating with other gtk4 ports.

Installing a flatpak already installed:
Screenshot from 2022-07-24 10 08 13

Installing a corrupt flatpakref, after opening and closing the "Details" widget. (#143 was merged to test this)
Screenshot from 2022-07-24 10 01 29

In general the the PR works though.

**Comment by Exalm** [comment]
It's 100% not the PR - a GTK issue, and probably an X11-specific one too.

File: src/Views/AbstractView.vala (hunk 1 above)
Comment (danrabbit) [reply src/Views/AbstractView.vala:62]: No, I don't have a good reason. I was probably just trying to keep diff size small. Updated!

**Review by jeremypw — APPROVED** [review]

**Pull request merged**